remarkapy/api.py

```diff
@@ -206,16 +206,157 @@ def list_documents(self) -> list[Entry]:
         if not response.status_code == 200:
             raise RemarkableAPIError(
                 f"Request failed with status code {response.status_code} when listing documents: {response.text}"
             )
-        results = response.json()
-        return parse_entries(results, fail_method="warn")
+
+        # NOTE we could store this hash somewhere as it probably changes only
+        # when a folder/file gets created/modified (if we decide to cache some data)
+        root_hash = response.json()['hash']
+
+        # List doc hashes on root folder
+        url = URLS.GET_FILE + root_hash
+        response = self._get(url, headers=headers)
+        obj_list = response.text.splitlines()
+
+        root = {}
+
+        # Add a file or a folder to a specific parent_hash or to root
+        def add_item_to_folder(folders, folder_hash: str, parent_hash: str = "", visible_name: str = "", metadata: dict = None, files: list = None):
+
+            # Trash is a special folder which doesn''t have a specific hash, just "trash"
+            if parent_hash == "trash":
+
+                if "trash" not in folders:
+                    folders["trash"] = {
+                        "visibleName": "Trash",
+                        "subfolders": {},
+                        "docs": {}
+                    }
+
+            # Recursively find or create a folder
+            def find_or_create_folder(folders, target_hash, parent_hash, visible_name):
+                # If no parent, add at root level
+                if not parent_hash:
+                    if target_hash not in folders:
+                        folders[target_hash] = {
+                            'name': visible_name,
+                        }
+
+                        if not metadata:
+                            folders[target_hash]['subfolders'] = {}
+
+                    return folders[target_hash]
+
+                for existing_folder_hash, folder_data in folders.items():
+
+                    if existing_folder_hash == parent_hash:
+                        if target_hash not in folder_data['subfolders']:
+                            folder_data['subfolders'][target_hash] = {
+                                'name': visible_name,
+                            }
+                            if not metadata:
+                                folder_data['subfolders'][target_hash]['subfolders'] = {
+                                }
+
+                        return folder_data['subfolders'][target_hash]
+
+                    if 'subfolders' in folder_data:
+                        found_folder = find_or_create_folder(
+                            folder_data['subfolders'], target_hash, parent_hash, visible_name)
+                        if found_folder:
+                            return found_folder
+
+                return None
+
+            current_folder = find_or_create_folder(
+                folders, folder_hash, parent_hash, visible_name)
+
+            print(
```

Comment:
probably makes sense to move this stuff into a `Client#debug()` call so that we can do something like, 

```python
c = Client(..., debug=True)
c.do_something() # shows logger output

c = Client(...) # defaults to debug=False)
c.do_something() # no output
```

that way we don't do anything to stdout without permission from the caller

Reply:
(can of course be a later change)

Reply:
Absolutely agree with you. We could also implement a verbose flag in the future. I can imagine it being particularly useful to power users :)